client/src/screens/BallotScreen.tsx

```diff
@@ -1,77 +1,147 @@
-import React, { useContext } from 'react'
-import { useParams } from 'react-router-dom'
+import React, { useContext, useState } from 'react'
+import { useParams, useHistory } from 'react-router-dom'
 import {
   getBallotStyle,
   getContests,
   getPrecinctById,
+  getElectionLocales,
 } from '@votingworks/ballot-encoder'
 import pluralize from 'pluralize'
 
 import { BallotScreenProps } from '../config/types'
 import AppContext from '../contexts/AppContext'
 
-import Button from '../components/Button'
+import Button, { SegmentedButton } from '../components/Button'
 import HandMarkedPaperBallot from '../components/HandMarkedPaperBallot'
 import { Monospace } from '../components/Text'
-import { getBallotFileName } from '../utils/election'
+import {
+  getBallotFileName,
+  getHumanBallotLanguageFormat,
+} from '../utils/election'
 import NavigationScreen from '../components/NavigationScreen'
 import HorizontalRule from '../components/HorizontalRule'
+import { DEFAULT_LOCALE } from '../config/globals'
+import { routerPaths } from '../components/ElectionManager'
 
 const BallotScreen = () => {
-  const { precinctId, ballotStyleId } = useParams<BallotScreenProps>()
+  const history = useHistory()
+  const {
+    precinctId,
+    ballotStyleId,
+    localeCode: currentLocaleCode,
+  } = useParams<BallotScreenProps>()
   const { election: e, electionHash } = useContext(AppContext)
   const election = e!
+  const availableLocaleCodes = getElectionLocales(election, DEFAULT_LOCALE)
+
   const precinctName = getPrecinctById({ election, precinctId })?.name
   const ballotStyle = getBallotStyle({ ballotStyleId, election })!
-
   const ballotContests = getContests({ ballotStyle, election })
 
+  const [isLiveMode, setIsLiveMode] = useState(true)
+  const toggleLiveMode = () => setIsLiveMode((m) => !m)
+  const changeLocale = (localeCode: string) =>
+    history.replace(
+      localeCode === DEFAULT_LOCALE
+        ? routerPaths.ballotsView({ precinctId, ballotStyleId })
+        : routerPaths.ballotsViewLanguage({
+            precinctId,
+            ballotStyleId,
+            localeCode,
+          })
+    )
+
+  const filename = getBallotFileName({
+    ballotStyleId,
+    election,
+    electionHash,
+    precinctId,
+    localeCode: currentLocaleCode || DEFAULT_LOCALE,
+  })
+
+  const printBallot = async () => {
+    const documentTitle = document.title
+    document.title = filename
+    await (window.kiosk ?? window).print()
```

Comment:
Is `window.print` an actual async call? I know (think?) `kiosk.print` is, but wondering if there's anything weird about explicitly `await`ing a blocking `window.print` call.

Reply:
It's harmless. It just boils down to `await undefined`, which is a no-op.